tests/server/server.spec.ts

```diff
@@ -31,15 +59,33 @@ describe('Server', () => {
 
   beforeEach(() => {
     server = new Server(port, host)
+    hasMockClientConnected = false
+    logMock = jest.spyOn(console, 'log').mockImplementation()
   })
 
-  test('logs a message to the terminal when a connection establishes', () => {
-    logMock = jest.spyOn(console, 'log').mockImplementation()
+  afterEach(() => {
+    logMock.mockRestore()
+  })
+
+  test('logs a message when the TCP server is open for connections', () => {
     server.openTCPConnection()
     expect(logMock).toHaveBeenCalledTimes(1)
     expect(logMock).toHaveBeenCalledWith(
       `TCP server at http://${host}:${port}/.`
     )
-    logMock.mockRestore()
+  })
+
+  test('logs a message when a TCP connection is established', () => {
+    hasMockClientConnected = true
+    server.openTCPConnection()
+    expect(logMock).toHaveBeenCalledTimes(2)
+    expect(logMock).toHaveBeenCalledWith('Client has connected.')
+  })
+
+  test('responds to a client connected to the TCP server', () => {
+    hasMockClientConnected = true
+    server.openTCPConnection()
+    expect(server.socket.write).toHaveBeenCalledTimes(1)
+    expect(server.socket.write).toHaveBeenCalledWith('Connection established.')
```

Comment:
🎉 

nitpick: Instead of using `toHaveBeenCalledTimes(1)` you can use `toHaveBeenCalled()`

Reply:
Thank you! I went ahead and updated the matcher! :)